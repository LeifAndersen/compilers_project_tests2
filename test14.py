x = {}
x = []
x = ()
def a(): pass
def a(): lambda: 3
a()()
z = (1)
z = (1,)
z = (1,1)
z = (1,1,)
z = (1,1,1)
y = [1]
y = [1,]
y = [1,1]
y = [1,1,]
y = [1,1,1]
y=[]
def a(a,): pass
def a(a): pass
def a(a,a,a): pass
for i in [1,23,4,6]:
 for j in [1,2]:
      for k in range(50):
        print(i*j*k);

