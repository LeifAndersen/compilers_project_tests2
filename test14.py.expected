(program
 (= (x) (dict))
 (= (x) (list))
 (= (x) (tuple))
 (def (a) (pass))
 (def (a) (expr (lambda () 3)))
 (expr (indexed a (called) (called)))
 (= (z) 1)
 (= (z) (tuple 1))
 (= (z) (tuple 1 1))
 (= (z) (tuple 1 1))
 (= (z) (tuple 1 1 1))
 (= (y) (list 1))
 (= (y) (list 1))
 (= (y) (list 1 1))
 (= (y) (list 1 1))
 (= (y) (list 1 1 1))
 (= (y) (list))
 (def (a a) (pass))
 (def (a a) (pass))
 (def (a a a a) (pass))
 (for
  i
  (list 1 23 4 6)
  (suite
   (for
    j
    (list 1 2)
    (suite
     (for
      k
      (indexed range (called 50))
      (suite (expr (indexed print (called (term i ("*" j) ("*" k))))))))))))

